rust/connlib/snownet/src/node.rs

```diff
@@ -514,7 +508,7 @@ where
                     .current_candidates()
                     .filter(|c| c.kind() == CandidateKind::Relayed)
                 {
-                    agent.invalidate_candidate(&candidate);
+                    remove_local_candidate(id, agent, &candidate, &mut self.pending_events);
```

Comment:
This fix here could be isolated into its own PR. It is an oversight where we don't send the invalidated relay candidate to the remote when we migrate from one relay to another.